src/figuresClasses.ts

```diff
@@ -1,19 +1,77 @@
-export interface Figure {
+type Color = 'red' | 'green' | 'blue';
+
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 
 }
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
 export class Triangle implements Figure {
+  shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('your error message');
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in error message.
![image](https://github.com/mate-academy/ts_figures_classes/assets/63243136/eb737714-fa8d-4eb1-bd4c-788e43ea196a)